### PR TITLE
Added function for the same

### PR DESCRIPTION
**Pull request by avinilcode**
Please let review and let me know

### CODE REVIEW

File: quickstart/docker/image/ziti-cli-functions.sh [hunk 1]
```diff
@@ -1471,4 +1471,17 @@ function getFileOverwritePermission() {
   fi
 }
 
+function isZitiEdgeControllerPortListening(){
```
Comment (dovholuknf) [inline quickstart/docker/image/ziti-cli-functions.sh:1474]: let's make this a more generic "isPortAvailable" that takes a parameter of the port to check. Probabaly need to turn that logic around then as well and need to pass the port to be checked and not hardcode the 1280.  Also need to add this to the `checkPrereqs` function. where it does something like check "if !isPortAvailable print error and exit"

Comment (avinilcode) [reply quickstart/docker/image/ziti-cli-functions.sh:1474]: ok

Comment (avinilcode) [reply quickstart/docker/image/ziti-cli-functions.sh]: Please review now

**Comment by gberl002** [comment]
@avinilcode Thanks for working on this for us, this is a feature I've been wanting but haven't had time to implement it so I'm excited to see this work.

However, this doesn't seem to work for me. I tried on MacOS 12.5 and on Ubuntu 20.04. The first issue was that the syntax for the isPortAvailable call `!isPortAvailable: command not found`. I added a space between the exclamation and function call and that fixed that issue.

However, after that fix, I 'm seeing this output on both mac and linux
```
sudo lsof -i:1280: command not found
[0: command not found
Port 1280 is not available hence exiting
```
So, it seems that it doesn't like this line `if ["${#ports[*]}" == 0]` and I'm not sure why the first line as that should work and it does when running that command straight from the terminal.

Have you tested this locally to confirm it works for you and, if so, what OS are you testing this with?

**Comment by avinilcode** [comment]
> @avinilcode Thanks for working on this for us, this is a feature I've been wanting but haven't had time to implement it so I'm excited to see this work.
> 
> However, this doesn't seem to work for me. I tried on MacOS 12.5 and on Ubuntu 20.04. The first issue was that the syntax for the isPortAvailable call `!isPortAvailable: command not found`. I added a space between the exclamation and function call and that fixed that issue.
> 
> However, after that fix, I 'm seeing this output on both mac and linux
> 
> ```
> sudo lsof -i:1280: command not found
> [0: command not found
> Port 1280 is not available hence exiting
> ```
> 
> So, it seems that it doesn't like this line `if ["${#ports[*]}" == 0]` and I'm not sure why the first line (the lsof not found) as that should work and it does when running that command straight from the terminal.
> 
> Have you tested this locally to confirm it works for you and, if so, what OS are you testing this with?

@gberl002 I am not able to run this, i tried the following "./ziti-cli-functions.sh isPortAvailable 1280" but there no output. Can you tell how to run it. And thanks for testing on my behalf.

**Comment by gberl002** [comment]
@avinilcode since you were unable to run this locally, I went ahead and added this on a separate branch. That branch has since been merged so this is no longer needed. Thank you for taking a look at this.

**Pull request closed without merging**